### PR TITLE
Remove trickle down router props

### PR DESCRIPTION
**Pull request by cstns**
Remove trickle down router props and cohesive way of using team, teamMembership and isVisitingAdmin

## Description

Focusing mainly on teamMembership & team props which can be used directly from the permissionsMixin, or using accessing the account store directly where the permission mixin is not used. 

The isVisitingAdmin computed prop has also been moved in the permissionsMixin

## Related Issue(s)
none

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by joepavitt** [comment]
I won't be able to review this today, so if someone else can step in, would appreciate it. My main point of concern is that if a user is logged in and viewing Team 1, and they directly navigate in the browser to a Team 2 page (e.g. viewing instances) that the new view reflects the _correct_ team's data

**Comment by cstns** [comment]
I added two new tests to cover these specific scenarios

File: frontend/src/pages/application/createInstance.vue [hunk 1]
```diff
@@ -68,7 +69,7 @@ export default {
         NavItem,
         SideNavigation
     },
-    mixins: [applicationMixin],
+    mixins: [applicationMixin, permissionMixin],
```
Comment (joepavitt) [inline frontend/src/pages/application/createInstance.vue:72]: I can't see where this is used in this component? We use the `team`, but that's already imported in the `mapState` directly, so need for the mixin here.

**Review by joepavitt — CHANGES_REQUESTED** [review]
You're missing an opportunity for the `permissionsMixin` in `pages/application/Overview.vue`

**Review by Steve-Mcl — APPROVED** [review]
@joepavitt I went through this line by line with Serban . We found 1 left over prop and fixed that.  

did a full review on pre-staging, tested all pages touched (with dev tools console open) and direct url access to alt teams etc. 

I am happy to approve but will not merge as you may see something i dont

**Comment by joepavitt** [comment]
@Steve-Mcl thanks Steve, just left my own comments on this too

**Review by Steve-Mcl — APPROVED** [review]
all good from my pov.

File: frontend/src/pages/application/createInstance.vue (hunk 1 above)
Comment (cstns) [reply frontend/src/pages/application/createInstance.vue:72]: Good catch! It doesn't seem to be used anywhere

**Review by joepavitt — APPROVED** [review]

**Pull request merged**